Units/parser-v.r/torture.d/expected.tags

```diff
@@ -0,0 +1,4 @@
+main	input.v	/^fn main() {$/;"	f	line:1	typeref:typename:	signature:()	roles:def	end:17
+a	input.v	/^	a := 1$/;"	v	line:3	fn:main	roles:def
+C.SYSTEM_INFO	input-1.v	/^struct C.SYSTEM_INFO {$/;"	s	line:1	roles:def	end:2
+C.foo	input-1.v	/^fn C.foo(bar string) i64$/;"	f	line:3	typeref:typename:i64	signature:(bar string)	roles:def
```

Comment:
Though we don't have consistent policy, but in some important parsers in ctags choose `foo` instead of `C.foo` when tagging this kind of item.

Let me show an example from C++:

```
$ cat /tmp/Input.cpp 
class C {
	void f(void);
};

void C::f(void)
{
	return;
}
$ ctags -o - Input.cpp
C	Input.cpp	/^class C {$/;"	c	file:
f	Input.cpp	/^void C::f(void)$/;"	f	class:C	typeref:typename:void

$ ctags -o - --extras=+q Input.cpp
C	Input.cpp	/^class C {$/;"	c	file:
C::f	Input.cpp	/^void C::f(void)$/;"	f	class:C	typeref:typename:void
f	Input.cpp	/^void C::f(void)$/;"	f	class:C	typeref:typename:void
```

The parser doesn't make a tag for `C::f` by default.
`C::f` is tagged only when `--extras=+q` option is passed.

Reply:
@masatake Thanks! I'll look in to making making the V parser respect the `q` extra. 👍  For some modules/objects, I may even be able to work out the type of the qualifying symbol from CORK, I think?

Reply:
I will explain in a language independent way.

Consider a parser reads "C.f()" in input.foo.

We recognize this as "a function f is defined in C".
How about ctags?

<D>If C is defined in the input.foo like: "class C {...}", ctags may know the corkId for C.
When making a tag for `f`, put the cork id of C into the scope field of `f`. See `
scopeIndex` in sTagEntryInfo in main/entry.h. You may know this.


<R> What I would like to write is about the case when C is not defined in the input.foo.
(Though I don't know there is such a case in V language.)

In such a case, make a reference tag for C when parsing `C.f`, and get a cork ID for `C`.
Once, you get a cork ID for C, what you have to do is the same as <D>.
In some languages, we can know the kind for C in "C.f()". e.g. namespace, class, struct, etc. For the case, you reserved "Y":
```
{true, 'Y', "unknown", "unknown, imported symbols",
	 .referenceOnly = false, ATTACH_ROLES (VUnknownRoles)},
```
.

I used this technique in parsers/python.c and parsers/jscript.c.
See also 19f653c46e and https://docs.ctags.io/en/latest/man/ctags-lang-python.7.html.

Reply:
I think this is fixed now.

`torture.d/expected.tags`
```
main	input.v	/^fn main() {$/;"	f	line:1	typeref:typename:	signature:()	roles:def	end:17
a	input.v	/^	a := 1$/;"	v	line:3	fn:main	roles:def
SYSTEM_INFO	input-1.v	/^struct C.SYSTEM_INFO {$/;"	s	line:1	external:C	roles:def	end:2
foo	input-1.v	/^fn C.foo(bar string) i64$/;"	f	line:3	external:C	typeref:typename:i64	signature:(bar string)	roles:def
```